From 518b51265928bc50d36e4814a050fe9cd66795c8 Mon Sep 17 00:00:00 2001
From: Billy Tsai <billy_tsai@aspeedtech.com>
Date: Wed, 7 Sep 2022 15:24:05 +0800
Subject: [PATCH 17/30] i3c: aspeed: Read rx fifo when tid == 0x8 in slave mode

In slave mode, tid will only be 0x8 (master) or 0xf (DEFSLAVE CCC)
If it is 0xf the data will auto update to the DCR and rx fifo won't get
any data. So, we only need to read rx fifo when tid == 0x8

Signed-off-by: Billy Tsai <billy_tsai@aspeedtech.com>
Change-Id: I7cafa54d1522969f3cabe1bafafa71e33c84fec9
---
 drivers/i3c/i3c_aspeed.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/i3c/i3c_aspeed.c b/drivers/i3c/i3c_aspeed.c
index f2ff191f25..51decfb743 100644
--- a/drivers/i3c/i3c_aspeed.c
+++ b/drivers/i3c/i3c_aspeed.c
@@ -144,6 +144,8 @@ union i3c_device_resp_queue_port_s {
 		volatile uint32_t err_status : 4;		/* bit[31:28] */
 	} fields;
 }; /* offset 0x10 */
+#define SLAVE_TID_MASTER_WRITE_DATA 0x8
+#define SLAVE_TID_DEFSLV_WRITE_DATA 0xF
 
 union i3c_ibi_queue_status_s {
 	volatile uint32_t value;
@@ -686,7 +688,8 @@ static void i3c_aspeed_slave_rx_data(struct i3c_aspeed_obj *obj)
 		struct i3c_slave_payload *payload;
 
 		resp.value = i3c_register->resp_queue_port.value;
-		if (resp.fields.data_length && !resp.fields.err_status) {
+		if (resp.fields.data_length && !resp.fields.err_status &&
+		    resp.fields.tid == SLAVE_TID_MASTER_WRITE_DATA) {
 			if (cb->write_requested) {
 				payload = cb->write_requested(obj->slave_data.dev);
 				payload->size = resp.fields.data_length;
-- 
2.24.1

